ledger/src/ui/src/bolos_ux_handlers.c

```diff
@@ -0,0 +1,171 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+#include "bolos_ux_common.h"
+#include "bootloader.h"
+
+volatile unsigned char autoexec = 0;
+
+// run the signer application
+static void run_signer_app(void) {
+    unsigned int i = 0;
+    while (i < os_registry_count()) {
+        application_t app;
+        os_registry_get(i, &app);
+        if (!(app.flags & APPLICATION_FLAG_BOLOS_UX)) {
+            if (is_authorized_signer(app.hash)) {
+                G_bolos_ux_context.app_auto_started = 1;
+                screen_stack_pop();
+                io_seproxyhal_disable_io();
+                os_sched_exec(i); // no return
+            }
+        }
+        i++;
+    }
+}
+
+/**
+ * BOLOS_UX_BOOT_ONBOARDING handler
+ *
+ * Shows onboarding screen and waits for commands if device is not onboarded,
+ * does nothing otherwise.
+ *
+ * @ret BOLOS_UX_OK if device is already onboarded, never returns if an actual
+ *      onboarding was performed
+ */
+unsigned int handle_bolos_ux_boot_onboarding() {
+    // re apply settings in the L4 (ble, brightness, etc) after exiting
+    // application in case of wipe
+    screen_settings_apply();
+
+    // request animation when dashboard has finished displaying all the
+    // elements (after onboarding OR the first time displayed)
+    G_bolos_ux_context.dashboard_redisplayed = 1;
+
+    // avoid reperso is already onboarded to avoid leaking data through
+    // parameters due to user land call
+    if (os_perso_isonboarded()) {
+        return BOLOS_UX_OK;
+    }
+
+    io_seproxyhal_init();
+    USB_power(1);
+    screen_settings_apply();
+    screen_not_personalized_init();
+    bootloader_main(BOOTLOADER_MODE_ONBOARD);
+    // bootloader_main() actually never returns when onboarding mode is active,
+    // so this value is never actually returned to the caller
+    return BOLOS_UX_CANCEL;
+}
+
+/**
+ * BOLOS_UX_DASHBOARD handler
+ *
+ * Shows dashboard screen when autoexec == 0, or loads signer app when
+ * autoexec == 1
+ */
+void handle_bolos_ux_boot_dashboard() {
+    // apply settings when redisplaying dashboard
+    screen_settings_apply();
+
+    // when returning from application, the ticker could have been
+    // disabled
+    io_seproxyhal_setup_ticker(100);
+    // Run signer application once
+    if (autoexec) {
+        autoexec = 0;
+        run_signer_app();
```

Comment:
Shall we also add a `return` after this to prevent the following line from running?